Makefile

```diff
@@ -1,5 +1,4 @@
-v ?= $(shell git describe --tags --abbrev=0)
-IMAGEVERSION ?= $(v:v%=%)
+BUILDVERSION := $(shell grep "appVersion" helm/charts/infra/Chart.yaml | sed -e "s/.*: //")
```

Comment:
Not something we can change right now, but this is a great example of why Makefiles are such a bad fit for this use case of "automating development tasks" (despite them being super widely used for that purpose).

Every single invocation of `make` will have to run this grep, even though most won't make use of it. I don't think there's any way around that for make.

Over time the number of variables builds up, and we end up with a very slow `make` because it's a bad fit for the job. 

An equivalent bash script would be able to calculate this only for the task that requires it.

Reply:
While this is true, there are method to delay execution until the value is needed. In this case, using `$$(...)` instead of `$(shell ...)` assigns `BUILDVERSION` to the command itself rather than the output. When the line referencing `BUILDVERSION` is executed, it will create a subshell and run the `grep`. 

This is actually used later on in `make dev` where the image SHA is evaluated _after_ the image is built rather than when `make` is called which is what `$(shell ...)` will. In that case, using `$(shell ...)` will not have the right output, the annotation is unlikely to change, and the pod will not automatically restart

Reply:
Cool, I did not know about this feature of make!  I still think that an equivalent bash script is going to be easier for most people to figure out, but good to know that there are ways to defer execution with `make`.